system/jlib/jsecrets.cpp

```diff
@@ -719,30 +722,39 @@ extern jlib_decl bool getSecretValue(StringBuffer & result, const char *category
     return true;
 }
 
-bool getSecretUdpKey(MemoryAttr &updkey)
+void initSecretUdpKey()
 {
-    bool ret = false;
-    updkey.clear();
+    if (udpKeyInitialized)
+        return;
+
 #if defined(_CONTAINERIZED) && defined(_USE_OPENSSL)
     BIO *in = BIO_new_file("/opt/HPCCSystems/secrets/certificates/udp/tls.key", "r");
     if (in == nullptr)
-        return false;
+        return;
     EC_KEY *eckey = PEM_read_bio_ECPrivateKey(in, nullptr, nullptr, nullptr);
     if (eckey)
     {
         unsigned char *priv = NULL;
         size_t privlen = EC_KEY_priv2buf(eckey, &priv);
         if (privlen != 0)
         {
-            updkey.set(privlen, priv);
+            udpKey.set(privlen, priv);
             OPENSSL_clear_free(priv, privlen);
-            ret = true;
         }
         EC_KEY_free(eckey);
     }
     BIO_free(in);
 #endif
-    return ret;
+    udpKeyInitialized = true;
+}
+
+const MemoryAttr &getSecretUdpKey(bool required)
+{
+    if (!udpKeyInitialized)
+        throw makeStringException(-1, "UPD Key not initialized.");
```

Comment:
Is this supposed to say UDP ?

Reply:
My fingers always seem to like this combination better.  Will change.